Update `identity-account` description (#887)
iotaledger/identity.rs: identity-account/README.md

> # IOTA Identity - Account

Reviewer: There are still `cargo doc` warnings.

E.g.

```rust
warning: unresolved link to `crypto`
  --> identity-account\src\error.rs:21:34
   |
21 |   /// Caused by errors from the [crypto] crate.
   |                                  ^^^^^^ no item named `crypto` in scope
```

Run `cargo doc --no-deps --open` to see and fix them.

Author: I fixed all the account related ones.

The [`Account`](crate::account::Account) is an interface for creating and managing identities on the IOTA Tangle, handling publishing and secure storage automatically. It provides convenience functions such as:

- Creating and publishing a new IOTA DID.
- Updating DID Document contents:
  - Verification Methods.
  - Verification Relationships.
  - Services.
- Managing private cryptographic keys securely.
- Signing credentials.
- Encrypting messages.

## Account Creation

Creating an [`Account`](crate::account::Account) is done through the [`AccountBuilder`](crate::account::AccountBuilder).

```{.ignore .rust}
  let account: Account = Account::builder()
    .create_identity(IdentitySetup::default())
    .await?;
```

> ## Operations
> 
> Operations can be done by calling their specific function on the account, for instance:

Reviewer: ```suggestion
## Update Operations

Updating a DID Document can be performed through the [`update_identity`](crate::account::Account::update_identity) function on the [`Account`](crate::account::Account). For example, adding a new verification method to the DID Document:
```

> ```{.ignore .rust}
>   account
>     .update_identity()
>     .create_method()
>     .content(MethodContent::GenerateEd25519)
>     .fragment("my-next-key")
>     .apply()
>     .await?;

Reviewer: ```suggestion
```rust,ignore
account
  .update_identity()
  .create_method()
  .content(MethodContent::GenerateEd25519)
  .fragment("my-next-key")
  .apply()
  .await?;
```

# Iota Identity Account

> The Account is an interface for creating and managing identities on the Iota Tangle. It includes easy-to-use functions that allow many operations including:

Reviewer: ```suggestion
The [`Account`](crate::account::Account) is an interface for creating and managing identities on the IOTA Tangle, handling publishing and secure storage automatically. It provides convenience functions such as:
```

> - Creating an identity or resolving an existing DID Document.
> - Updating a DID Document by adding or removing:
>   - Services.
>   - Verification Methods.
>   - Verification Relationships.
>   - Custom properties.
> - Managing and storing the state of DID Documents.
> - Managing and storing private encryption keys used as verification methods.

Reviewer: ```suggestion
- Creating and publishing a new IOTA DID.
- Updating DID Document contents:
  - Verification Methods.
  - Verification Relationships.
  - Services.
- Managing private cryptographic keys securely.
- Signing credentials.
- Encrypting messages.
```

## Account Creation

Creating an account is done through the `AccountBuilder`.

```rust
  let account: Account = Account::builder()
    .create_identity(IdentitySetup::default())

>     .await?;

Reviewer: Or go with https://github.com/iotaledger/identity.rs/pull/887#discussion_r896639318

> ```rust
>   let account: Account = Account::builder()
>     .create_identity(IdentitySetup::default())
>     .await?;
> ```

Reviewer 2: This will run on `cargo test` and might occasionally fail because of some io error. One can work around this with the [Easy doc initialization pattern](https://rust-unofficial.github.io/patterns/idioms/rustdoc-init.html). 

```suggestion
use identity_account::Account;
use identity_account::Result;
 
async fn build_account() -> Result<Account> {
    let account: Account = Account::builder()
    .create_identity(IdentitySetup::default())
    .await?;
    Ok(account)
```

This way the `cargo test` checks that the code compiles, but it is not run.

> Some options can be set for the `AccountBuilder` including:

Reviewer 3: ```suggestion
Various options can be set for the `AccountBuilder` including:
```

- `autosave`: sets the account auto-save behaviour.

>   - `AutoSave::Every` => save to storage on every update

Reviewer 3: ```suggestion
  - `AutoSave::Every` => save to storage on every update.
```

>   - `AutoSave::Every` => save to storage on every update.

Reviewer 2: Would be great if you provided links to items i.e. [`AutoSave`](crate::account::AutoSave)

>   - `AutoSave::Never` => never save to storage when updating

Reviewer 3: ```suggestion
  - `AutoSave::Never` => never save to storage when updating.
```

  - `AutoSave::Batch` => save to storage after every `n` updates.
- `autopublish`: sets the account auto-publish behaviour.

>   - `true` => publish to the Tangle on every DID document change

Reviewer 3: ```suggestion
  - `true` => publish to the Tangle on every DID document change.
```

>   - `false` => never publish automatically

Reviewer 3: ```suggestion
  - `false` => never publish automatically.
```

> Various options can be set for the `AccountBuilder` including:
> 
> - `autosave`: sets the account auto-save behaviour.
>   - `AutoSave::Every` => save to storage on every update.
>   - `AutoSave::Never` => never save to storage when updating.
>   - `AutoSave::Batch` => save to storage after every `n` updates.
> - `autopublish`: sets the account auto-publish behaviour.
>   - `true` => publish to the Tangle on every DID document change.
>   - `false` => never publish automatically.
> - `storage`: sets the account storage adapter.
> - `client`: sets the IOTA Tangle `Client`, this determines the `Network` used by the identity.
> - `create_identity`: creates a new identity based on the builder configuration and returns an `Account` instance to manage it.
> - `load_identity`: loads an existing identity with the specified `did` using the current builder configuration. The identity must exist in the configured `Storage`.

Reviewer: Try not to duplicate code definitions, use intradoc links at most. Or just include a code example similar to the `config` example to demonstrate some of the available options.

```suggestion
```

Author: It's clear it has maintainability issues, but It's either duplicating the examples or duplicating the function definitions, I think having the code definitions in the README gives a quick overview about the possible options that can be done with the account.

Reviewer: We have the proper examples for a reason.

They can similarly see the range of functions if they click on the `Account` intradoc link.

The only thing obscure is really the update functions, so those should probably be linked to via `IdentityUpdater`.

## Operations

> Operations can be done by calling their specific funciton on the account, for instance:

Reviewer 3: ```suggestion
Operations can be done by calling their specific function on the account, for instance:
```

  ```rust
  account
    .update_identity()
    .create_method()
    .content(MethodContent::GenerateEd25519)
    .fragment("my-next-key")
    .apply()
    .await?;
```

is used to add a verification method to the DID Document.

> Similar functions are exposed for different operations including:
> 
> - `delete_method`: deletes a verification method.
> - `attach_method_relationship`: attaches one or more verification relationships to a method.
> - `detach_method_relationship` detaches one or more verification relationships to a method.
> - `create_service`: adds a new Service to the DID Document.
> - `delete_service`: deletes a Service from the DID Document.
> - `set_controller`: sets the controllers of the DID document.
> - `set_also_known_as`: sets the `alsoKnownAs` property in the DID document.
> - `fetch_document`: fetches the latest changes from the tangle and **overwrites** the local document.
> - `encrypt_data`: encrypts the given `plaintext` with the specified `encryption_algorithm` and `cek_algorithm`.
> - `decrypt_data`: decrypts the given `data` with the key identified by `fragment` using the given `encryption_algorithm` and `cek_algorithm`.
> - `resolve_identity`: resolves the DID Document associated with this `Account` from the Tangle.
> - `delete_identity`: removes the identity from the local storage entirely.
> - `sign`: signs `data` with the key specified by `fragment`.
> - `publish`: pushes all unpublished changes to the tangle in a single message.
> - `update_document_unchecked`: overwrites the `IotaDocument` this account manages, **without doing any validation**.

Reviewer: Try not to duplicate function definitions and doc-comments. It would be quite difficult to maintain this otherwise.

```suggestion
```
